gdal/alg/gdalwarpkernel.cpp

```diff
@@ -6174,25 +6252,34 @@ static void GWKAverageOrModeThread( void* pData)
                                 dfBandDensity > BAND_DENSITY_THRESHOLD )
                             {
                                 nCount++;
-                                if( dfTotal > dfValueRealTmp )
+                                if( dfTotalReal > dfValueRealTmp )
+                                {
+                                    dfTotalReal = dfValueRealTmp;
+                                }
+                                if( bIsComplex && (dfTotalImag > dfValueImagTmp ))
                                 {
-                                    dfTotal = dfValueRealTmp;
+                                    dfTotalImag = dfValueImagTmp;
                                 }
                             }
                         }
                     }
 
                     if( nCount > 0 )
                     {
-                        dfValueReal = dfTotal;
+                        dfValueReal = dfTotalReal;
+                        if (bIsComplex)
+                        {
+                            dfValueImag = dfTotalImag;
+                        }
                         dfBandDensity = 1;
                         bHasFoundDensity = true;
                     }
                 }  // GRA_Min.
                 else if( nAlgo == GWKAOM_Quant )
                 // poWK->eResample == GRA_Med | GRA_Q1 | GRA_Q3.
                 {
-                    std::vector<double> dfValuesTmp;
+                    std::vector<double> dfRealValuesTmp;
```

Comment:
same here. what is the semantics of quantile for complex values...

Reply:
See above. If one were to use min/max for scaling operations - using max(maxReal, maxImag) and min(minReal, minImag) is probably better for quantization than using min/max of abs values.